### PR TITLE
Wellformed the table's rowspan and/or colspan attributes

### PR DESCRIPTION
**Pull request by TheNorthMemory**
- Setup GitHub workflows
- Test against with `markdown-it` v14
- optimize `.npmignore` only with necessary files
- Wellformed the table's rowspan and/or colspan attributes

### CODE REVIEW

**Comment by TheNorthMemory** [comment]
optimized some jsdocs and added some `@typedef` for better code tracing, see [here](https://github.com/arve0/markdown-it-attrs/pull/156/commits/3c0c5b389cae5df4068299389c7d321117c7e11c)

**Comment by TheNorthMemory** [comment]
Wellformed the table's _rowspan_ and/or _colspan_ attributes, usage sample below:

```md
| A                       | B   | C   | D                |
| ----------------------- | --- | --- | ---------------- |
| 1                       | 11  | 111 | 1111 {rowspan=3} |
| 2 {colspan=2 rowspan=2} | 22  | 222 | 2222             |
| 3                       | 33  | 333 | 3333             |

{border=1}
```

Output:
```html
<table border="1">
  <thead>
    <tr>
      <th>A</th>
      <th>B</th>
      <th>C</th>
      <th>D</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>1</td>
      <td>11</td>
      <td>111</td>
      <td rowspan="3">1111</td>
    </tr>
    <tr>
      <td colspan="2" rowspan="2">2</td>
      <td>22</td>
    </tr>
    <tr>
      <td>3</td>
    </tr>
  </tbody>
</table>
```

File: test.js [hunk 1]
```diff
@@ -49,6 +49,35 @@ function describeTestsWithOptions(options, postText) {
       const res = utils.getAttrs(replaceDelimiters(src, options), 0, options);
       assert.deepEqual(res, expected);
     });
+
+    it(replaceDelimiters('should parse attributes whose are ignored the key chars(\\t,\\n,\\f,\\s,/,>,",\',=) eg: {gt>=true slash/=trace i\\td "q\\fnu e\'r\\ny"=}', options), () => {
```
Comment (arve0) [inline test.js:53]: Why should one ignore these chars? What is the intended use case?

File: test.js [hunk 2]
```diff
@@ -49,6 +49,35 @@ function describeTestsWithOptions(options, postText) {
...
+
+    it(replaceDelimiters('should throw an error while calling `hasDelimiters` with an invalid `where` param', options), () => {
+      assert.throws(() => utils.hasDelimiters(0, options), { name: 'Error', message: /Should be "start", "end" or "only"/ });
+      assert.throws(() => utils.hasDelimiters('', options), { name: 'Error', message: /Should be "start", "end" or "only"/ });
+      assert.throws(() => utils.hasDelimiters(null, options), { name: 'Error', message: /Should be "start", "end" or "only"/ });
+      assert.throws(() => utils.hasDelimiters(undefined, options), { name: 'Error', message: /Should be "start", "end" or "only"/ });
+      assert.throws(() => utils.hasDelimiters('center', options)('has {#test} delimiters'), { name: 'Error', message: /expected 'start', 'end' or 'only'/ });
+    });
+
+    it('should escape html entities(&,<,>,") eg: <a href="?a&b">TOC</a>', () => {
```
Comment (arve0) [inline test.js:68]: I do not think this project to the escaping? Guess this is tested for in markdown-it?

File: test.js [hunk 3]
```diff
@@ -49,6 +49,35 @@ function describeTestsWithOptions(options, postText) {
...
+      const src = '<a href="a&b">TOC</a>';
+      const expected = '&lt;a href=&quot;a&amp;b&quot;&gt;TOC&lt;/a&gt;';
+      const res = utils.escapeHtml(src);
+      assert.deepEqual(res, expected);
+    });
+
+    it('should keep the origional input which is not contains(&,<,>,") char(s) eg: |a|b|', () => {
```
Comment (arve0) [inline test.js:75]: Not sure why this test is necessary?

**Review by arve0 — COMMENTED** [review]
Hi 👋 Thanks for contributing :octocat: 

I must admit that the code is challenging, I do not know much about rowspan or colspan, and it has been some years since this code was written. I cannot totally grasp my own code either. The type-hinting will help, thank you for those!

If you could please answer my questions, I'll do a review soonish.

Thanks again 🙌

**Comment by TheNorthMemory** [comment]
> Hi 👋 Thanks for contributing :octocat:
> 
> I must admit that the code is challenging, I do not know much about rowspan or colspan, and it has been some years since this code was written. I cannot totally grasp my own code either. The type-hinting will help, thank you for those!
> 
> If you could please answer my questions, I'll do a review soonish.
> 
> Thanks again 🙌

The [rowspan](https://developer.mozilla.org/en-US/docs/Web/API/HTMLTableCellElement/rowSpan) and [colspan](https://developer.mozilla.org/en-US/docs/Web/API/HTMLTableCellElement/colSpan) are all the html standard for rending `table`. And this pkg was already supporting parsing and transform those as well. It's great. Just need a little bit extra modification, the [definition](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/td#usage_notes
) is:

picture from MDN


This improvement is solving the overflow cells, better for rendering. See the above notes.

File: test.js (hunk 1 above)
Comment (TheNorthMemory) [reply test.js:53]: When I run `npm test` and saw that were prompt:

File         | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
-------------|---------|----------|---------|---------|------------------------------------
All files    |    93.1 |    87.15 |   88.63 |   93.98 |
 index.js    |   87.34 |    80.64 |   78.57 |   90.32 | 117-124,138
 patterns.js |     100 |    95.23 |     100 |     100 | 327
 utils.js    |   92.12 |    89.58 |   83.33 |   91.86 | 81,145,198,237,241,253,270,274-277

This test case is covered the utils.js:L81 and reminded by the notes on L8-9:

https://github.com/arve0/markdown-it-attrs/blob/80a8e5278fb336725bcdde14b7c93ec8fbaafc04/utils.js#L8-L9

File: test.js (hunk 2 above)
Comment (TheNorthMemory) [reply test.js:68]: Yes, I agree with you. This function `escapeHtml` wasn't used in this project. But It is `exports` for outer usage. This test case is only for coverd L274-277.

File: test.js (hunk 3 above)
Comment (TheNorthMemory) [reply test.js:75]: Same reason for covered utils.js:L274-277

**Comment by TheNorthMemory** [comment]
A real rendered markdown table (490 rows x 3 columns) is launched [here](https://wechatpay.js.org/openapi/). It looks expectation and no more time spent(from the GitHub actions). Hopes it may helpful.

**Comment by TheNorthMemory** [comment]
Hi there, feel free take a look this website https://wechatpay.im/openapi/, there were more than 790+ rendered pages with this feature. Hopes this pr may be merged. @arve0

**Comment by arve0** [comment]
Sorry for the delay. It's published with version 4.3.0.

**Pull request merged**